### DISCUSSION TITLE
Will migrating from MongoDB to PostgreSQL benefit us?

### DISCUSSION THREAD

**Discussion by vitaliysignly** [prisma/prisma#15792] [Q&A]
We are currently developing a new API and would like to know the best strategy going forward, with regards to our database.

Our existing NodeJS API currently uses MongoDB (hosted on Azure via CosmosDB) for storing and querying all of our data. We then query this data directly, using Mongoose, and feed the information back to the client.

The new app that we’re developing is also written in NodeJS, using the Express, Apollo, Prisma and MongoDB stack.
During initial testing of this new API & stack, we came across a few issues with CosmosDB and it’s MongoDB API (which Prisma uses to query the data). Some of our queries did not work at all.
To resolve this, we switched over to MongoDB Atlas, hosted in Azure - which seems to have mitigated all of our data retrieval problems.
The issue now is performance.

We’re trying to determine whether it would be beneficial for us to continue using MongoDB or perhaps try PostgreSQL instead. PostgreSQL seems to be better supported by Prisma and we understand that relational databases are generally a lot faster in terms of performance.

Is it worth investing time into copying our data across to PostgreSQL or generally speaking, should we expect the same performance results?

**Comment by nurul3101** [comment]
Hello @vitaliysignly 👋 

You can definitely try out using PostgreSQL, you might need to revisit your database architecture as you are moving from NoSQL to a SQL Database. From the performance aspect, I would recommend you migrate some of your data to a new PostgreSQL database and replicate the usual traffic and check for response times, please note that there are other aspects as well which would determine overall performance like you might need to do database tuning, add appropriate indexes, make sure that database and your application server are in the same regions etc.

**Reply by vladi-stevanovic** [reply]
👋 Hello @vitaliysignly did you perhaps have a chance to check Nurul's recommendation? 
Let us know if you still have any questions, or feel free to "Mark as answer" if this discussion is now solved! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
